### PR TITLE
Fix warnings found by idea IDE in the SDK dir.

### PR DESCRIPTION
**Pull request by bogdandrutu**
* Use lambdas or method references where possible
* Remove dead code
* Make members final where possible
* Use Objects.equals instead of manually implementing the same functionality

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by carlosalberto — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]
Thanks!

**Comment by jkwatson** [comment]
Looks like this needs a rebase.

File: sdk/all/src/jmh/java/io/opentelemetry/sdk/metrics/MetricsTestOperationBuilder.java [hunk 1]
```diff
@@ -23,102 +23,90 @@
...
-            }
-
-            @Override
-            public void performBound() {
-              boundMetric.add(5L);
-            }
-          };
-        }
+      meter -> {
+        return new Operation() {
```
Comment (anuraaga) [inline sdk/all/src/jmh/java/io/opentelemetry/sdk/metrics/MetricsTestOperationBuilder.java:27]: While we're addressing IDE warnings, what about the IDE suggestion of removing `return` too? Up to you though

**Review by anuraaga — APPROVED** [review]

File: sdk/all/src/jmh/java/io/opentelemetry/sdk/metrics/MetricsTestOperationBuilder.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/all/src/jmh/java/io/opentelemetry/sdk/metrics/MetricsTestOperationBuilder.java:27]: Did not find that, but I will follow-up

**Review by thisthat — APPROVED** [review]

**Comment by anuraaga** [comment]
Thanks!

**Pull request merged**